crypto/x509/x509_vfy.c

```diff
@@ -986,15 +1008,144 @@ static int check_revocation(X509_STORE_CTX *ctx)
     }
     for (i = 0; i <= last; i++) {
         ctx->error_depth = i;
-        ok = check_cert(ctx);
-        if (!ok)
-            return ok;
+#ifndef OPENSSL_NO_OCSP
+        if (ctx->param->flags & X509_V_FLAG_OCSP_CHECK) {
```

Comment:
The new behavior added here requires new test cases for cert verification with OCSP-based status checks.